en/guide/extending_csd.md

```diff
@@ -1,6 +1,121 @@
+- Second is the camera device that does not support gstreamer source element. Such type of class must implement the read function to return camera frame buffers. Gstreamer pipeline with appsrc element is created for such devices. The appsrc element is fed with frames returned by read function.
+
+### CameraParameters
+CameraParameters is the class to store and retrieve parameters and its value. Camera parameter is identified by a string and a corresponding integer ID. Parameter can be of any defined [types](http://mavlink.org/messages/common#MAV_PARAM_EXT_TYPE_UINT8). The parameter name, ID, type and its value(converted as string) is stored in the map data structure in the CameraParameters class. Few commonly used camera parameters are defined with name, ID and supported values. Nevertheless, developer can declare custom parameters as per their requirement.
+
+```cpp
+class CameraParameters {
+```
+#### 1. Extend CameraDevice Class
+To add support for new type of camera device in CSD, a custom class need to be derived from CameraDevice base class.
+
+Example: CameraDeviceGazebo is the class that extends CameraDevice class.
+
+```cpp
+class CameraDeviceGazebo final : public CameraDevice {
+```
+CameraDevice has some pure virtual functions (runtime binding) that must be implemented by the child class. Other methods can also be overloaded to provide the functionality.
+
+Example : Details on Gazebo class
+
+```cpp
+class CameraDeviceGazebo final : public CameraDevice {
+public:
+    CameraDeviceGazebo(std::string device);
+    ~CameraDeviceGazebo();
+    std::string getDeviceId();
+    int getInfo(struct CameraInfo &camInfo);
+    bool isGstV4l2Src();
+    int init(CameraParameters &camParam);
+    int uninit();
+    int resetParams(CameraParameters &camParam);
+    int setParam(CameraParameters &camParam, std::string param, const char *param_value,
+                 size_t value_size, int param_type);
+```
+#### 2. Detect Camera Device
+After having added a custom CameraDevice class, there has be a logic in place to detect the custom camera device. A CameraComponent will be created for every camera device detected. V4L2 camera devices are detected by scanning the linux device nodes /dev/video*. Gazebo camera is detected based on --enable-gazebo compile time flag. Configurations associated with the camera device is read from the conf file.
+
+```cpp
+// prepare the list of cameras in the system
+int CameraServer::detectCamera(ConfFile &conf)
+```
+Example: For Gazebo CameraDevice
+
+```cpp
+int CameraServer::detect_devices_gazebo(ConfFile &conf, std::vector<CameraComponent *> &camList)
+```
+
+#### 3. Instantiate Camera Device
+The CameraComponent instantiates and owns the custom CameraDevice object. The instantiation of the object is based on the string that identifies the camera device.
```

Comment:
So again, here we need to phrase in terms of what the user needs to do "Developer needs to update Camera code to properly create and instantiate a camera device ..."

Reply:
12. Change the phrase appropriately from "what is done" to "what developers should do"